src/cluster/backend/zeromq/ZeroMQ.cc

```diff
@@ -356,7 +349,10 @@ bool ZeroMQBackend::DoPublishLogWrites(const logging::detail::LogWriteHeader& he
 }
 
 void ZeroMQBackend::Run() {
-    util::detail::set_thread_name(zeek::util::fmt("zmq-%p", this));
+    char name[4 + 2 + 16 + 1]; // zmq-0x<8byte pointer in hex><nul>
```

Comment:
This is default-initializing the elements (i.e., they'll have random values), let's instead zero-initialize them.

```suggestion
    char name[4 + 2 + 16 + 1]{}; // zmq-0x<8byte pointer in hex><nul>
```

For me `printf("%p", this)` seems to return a a string with 12 values, so properly initializing the buffer would prevent us potentially reading unit memory. The docs for `set_thread_name` state

>  OS limitations typically truncate the name to 15 bytes maximum.

so a big part might be thrown away anyway 🤷.

Reply:
> seems to return a string of length 11

This will depend on the address of `this`, with 8 byte pointers this could be larger, but yes, the name is truncated by the OS.

> properly initializing the buffer would prevent us potentially reading uninit memory

`snprintf()` will null terminate (so the extra null assignment I added is superflous, actually) - so this shouldn't be a concern.